app/models/policy_version.rb

```diff
@@ -86,6 +86,50 @@ def before_save
     self.policy_sha256 = Digest::SHA256.hexdigest(policy_text)
   end
 
+  def after_save
+    log_versions_off_limit
+    delete_versions_off_limit
+  rescue => e
+    Rails.logger.error("Error while ensuring policy '#{id}' versions retention limit: #{e.inspect}")
+  end
+
+  def log_versions_off_limit
+    versions_off_limit.all.each do |policy_version|
+      Rails.logger.debug("Deleting policy version: #{policy_version.except(:policy_text, :policy_sha256, :finished_at)}")
+    end
+  end
+
+  def versions_off_limit
+    Sequel::Model.db[<<-SQL, resource_id, policies_version_limit, resource_id]
+       WITH
+      "ordered_versions" AS (SELECT * FROM "policy_versions" WHERE ("resource_id" = ?) ORDER BY "version" DESC LIMIT ?)
```

Comment:
in the following method you indented this like this. please align between the 2
```suggestion
    Sequel::Model.db[<<-SQL, resource_id, policies_version_limit, resource_id]
    WITH
      "ordered_versions" AS (SELECT * FROM "policy_versions" WHERE ("resource_id" = ?) ORDER BY "version" DESC LIMIT ?)
```